pkg/collector/upgrade/v0_61_0.go

```diff
@@ -0,0 +1,142 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package upgrade
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/collector/adapters"
+)
+
+func upgrade0_61_0(u VersionUpgrade, otelcol *v1alpha1.OpenTelemetryCollector) (*v1alpha1.OpenTelemetryCollector, error) {
+	if len(otelcol.Spec.Config) == 0 {
+		return otelcol, nil
+	}
+
+	otelCfg, err := adapters.ConfigFromString(otelcol.Spec.Config)
+	if err != nil {
+		return otelcol, fmt.Errorf("couldn't upgrade to v0.61.0, failed to parse configuration: %w", err)
+	}
+
+	// Search for removed Jaeger remote sampling settings. (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14163)
+	receiversConfig, ok := otelCfg["receivers"].(map[string]interface{})
```

Comment:
Any idea why we see here:
```golang
map[receivers:map[jaeger:map[protocols:map[grpc:<nil>] remote_sampling:map[endpoint:jaeger-collector:14250 strategy_file:/etc/strategy.json strategy_file_reload_interval:10s tls:map[insecure:true]]] jaeger/2:map[protocols:map[grpc:<nil>] remote_sampling:map[endpoint:jaeger-collector2:14250 strategy_file:/etc/strategy2.json strategy_file_reload_interval:15s tls:map[insecure:true]]]] service:map[pipelines:map[traces:map[exporters:[nop] receivers:[jaeger jaeger/2]]]]]
```

The returned data doesnt match `map[string]interface{}` since the key is different: `jaeger:map[protocols:map[grpc:<nil>`.

Here is the yaml that is provided:
```yaml
---
receivers:
  jaeger:
    protocols:
      grpc:
    remote_sampling:
      endpoint: "jaeger-collector:14250"
      tls:
        insecure: true
      strategy_file: "/etc/strategy.json"
      strategy_file_reload_interval: 10s
  jaeger/2:
    protocols:
      grpc:
    remote_sampling:
      endpoint: "jaeger-collector2:14250"
      tls:
        insecure: true
      strategy_file: "/etc/strategy2.json"
      strategy_file_reload_interval: 15s
service:
  pipelines:
    traces:
      receivers: [jaeger, jaeger/2]
      exporters: [nop]
```